### PR TITLE
react_dynamic-list-of-todos

### PR DESCRIPTION
**Pull request by georgiyevv**
[DEMO LINK](https://svyatoslavPy.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,94 @@
...
+import { User } from './types/User';
+
+enum StatusesSelect {
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
```
Comment (etojeDenys) [inline src/App.tsx:23]: looks redundant to keep two states for todos

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,94 @@
...
+  const [query, setQuery] = useState('');
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [statusSelect, setStatusSelect] = useState('All');
```
Comment (etojeDenys) [inline src/App.tsx:28]:
```suggestion
  const [statusSelect, setStatusSelect] = useState(StatusesSelect.All);
```
move this value to the enum

File: src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,94 @@
...
+
+      getUser(selectedTodo?.userId)
+        .then(data => {
+          setUser(data);
+        })
+        .finally(() => {
+          setLoadingUser(false);
+        });
+    }
+  }, [selectedTodo]);
```
Comment (etojeDenys) [inline src/App.tsx:60]: let's move all user related information to the Modal component

File: src/App.tsx [hunk 4]
```diff
@@ -1,14 +1,94 @@
...
+
+  useEffect(() => {
+    if (statusSelect === StatusesSelect.Active) {
+      setTodos(initialTodos.filter(item => !item.completed));
+    } else if (statusSelect === StatusesSelect.Completed) {
+      setTodos(initialTodos.filter(item => item.completed));
+    } else {
+      setTodos(initialTodos);
+    }
+  }, [statusSelect, initialTodos]);
```
Comment (etojeDenys) [inline src/App.tsx:70]: you can just do like this
```
const filteredTodos = getFilteredTodos(initialTodos, statusSelect)
```
and filter the array in that function

**Review by etojeDenys — CHANGES_REQUESTED** [review]
your demo link doesn't work, feel free to ask for some help about this task in the chat

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Consider changing the icon for selected todo 
Screenshot 2024-04-30 at 13 22 05

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Make sure to deploy your fix

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,100 +1,74 @@
...
+                onClick={() => onSelectedTodo(todo)}
+                data-cy="selectButton"
+                className="button"
+                type="button"
+              >
+                <span className="icon">
+                  <i
+                    className={cn('far', {
+                      'fa-eye': !selectedTodo,
+                      'fa-eye-slash': selectedTodo,
```
Comment (anastasiiavorobiova) [inline src/components/TodoList/TodoList.tsx:63]:
```suggestion
                      'fa-eye-slash': selectedTodo.id === todo.id,
```

**Review by anastasiiavorobiova — APPROVED** [review]
Great work!

**Pull request closed without merging**